src/main.rs

```diff
@@ -771,30 +743,33 @@ const APP: () = {
         }
     }
 
-    #[task(binds = SPI3, resources = [adc1, dac1, iir_state, iir_ch], priority = 2)]
-    fn spi3(c: spi3::Context) {
-        let output: u16 = {
-            let a: u16 = c.resources.adc1.read().unwrap();
-            let x0 = f32::from(a as i16);
-            let y0 =
-                c.resources.iir_ch[1].update(&mut c.resources.iir_state[1], x0);
-            y0 as i16 as u16 ^ 0x8000
-        };
+    #[task(binds=DMA1_STR3, resources=[adcs, dacs, iir_state, iir_ch], priority=2)]
+    fn adc_update(c: adc_update::Context) {
+        let (adc0_samples, adc1_samples) =
+            c.resources.adcs.transfer_complete_handler();
 
-        c.resources.dac1.send(output).unwrap();
-    }
+        let mut dac0: [u16; SAMPLE_BUFFER_SIZE] = [0; SAMPLE_BUFFER_SIZE];
+        let mut dac1: [u16; SAMPLE_BUFFER_SIZE] = [0; SAMPLE_BUFFER_SIZE];
```

Comment:
Should we use [core::mem::MaybeUninit](https://doc.rust-lang.org/core/mem/union.MaybeUninit.html) to save the time initializing these buffers since we write to them just below here?

Reply:
Actually, could we add a method to the DAC module that returns a reference to the buffer the DMA is not currently  using, so we could write directly into that buffer? This would save us the time of copying from these buffers into the internal DAC buffers in `dacs.next_data()`

Reply:
Great recommendation here, as it removes the last unnecessary data operation from the read->dsp->write standpoint. It has now been removed.